File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/Address.java
package kz.greetgo.sandbox.controller.model;


import kz.greetgo.sandbox.controller.model.enums.AddressType;

public class Address {
    public Long id;
    public Long client;
    public AddressType type;
    public String street;
    public String house;
    public String flat;
    public Boolean actual;

    public Address(Long id, Long client, Object type, String street, String house, String flat, Boolean actual) {
        this.id = id;
        this.client = client;
        this.type = AddressType.valueOf(type.toString());
        this.street = street;
        this.house = house;
        this.flat = flat;
        this.actual = actual;
    }
    public Address(){

    }

    @Override
    public String toString() {
        return "Address{" +
                "id=" + id +
                ", client=" + client +
                ", type=" + type +
                ", street='" + street + '\'' +
                ", house='" + house + '\'' +
                ", flat='" + flat + '\'' +
                ", actual=" + actual +
                '}';
    }
}

File: sandbox.client/src/model/Address.ts
import {Account} from "./Account";
import {AddressType} from "./AddressType";

export class Address {
  public id : number;
  public client: number;
  public type:AddressType;
  public street:string;
  public house:string;
  public flat:string;

  constructor(TYPE) {
    this.type = AddressType.REG == TYPE?AddressType.REG:AddressType.FACT;
    this.street = "";
    this.house = "";
    this.flat = "";
  }

  public static create(a : any): Address {
    const ret = new Address(a.type);
    ret.assign(a);
    return ret;
  }

  assign(a : any){
    this.id = a.id;
    this.type = AddressType.REG == a.type?AddressType.REG:AddressType.FACT;
    this.flat = a.flat;
    this.street = a.street;
    this.house = a.house;
    this.client = a.client;
  }
}
